### PR TITLE
Update XPATH README.md

### PR DESCRIPTION
**Pull request by Josar**
Added an example which adresses basic operation of the xpath transformation.

Signed-off-by: josar <josuaarndt@live.de>

### CODE REVIEW

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 1]
```diff
@@ -1,3 +1,52 @@
 # XPath Transformation Service
 
-Transform XML input using an [XPath expression](https://www.w3.org/TR/xpath/#section-Expressions).
+Transforms an [XML](https://www.w3.org/XML/) input using an [XPath](https://www.w3.org/TR/xpath/#section-Expressions) expression.
+
+## Advanced Example
+Given a retrived XML e.g. from an HIK Vision device with the namespace `xmlns="http://www.hikvision.com/ver20/XMLSchema"`.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: Given a retrieved XML (e.g. from an HIK Vision device with the namespace `xmlns="http://www.hikvision.com/ver20/XMLSchema"`.):

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 2]
```diff
@@ -1,3 +1,52 @@
...
+<?xml version="1.0" encoding="UTF-8"?>
+<PTZStatus version="2.0" xmlns="http://www.hikvision.com/ver20/XMLSchema">
+	<AbsoluteHigh>
+		<elevation>0</elevation>
+		<azimuth>450</azimuth>
+		<absoluteZoom>10</absoluteZoom>
+	</AbsoluteHigh>
+</PTZStatus>
+```
+A simple xpath query to fetch the Azimut value does not work as it does not adress the namespace.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: Needs a newline below end of code fence.

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 3]
```diff
@@ -1,3 +1,52 @@
 # XPath Transformation Service
 
-Transform XML input using an [XPath expression](https://www.w3.org/TR/xpath/#section-Expressions).
+Transforms an [XML](https://www.w3.org/XML/) input using an [XPath](https://www.w3.org/TR/xpath/#section-Expressions) expression.
+
+## Advanced Example
+Given a retrived XML e.g. from an HIK Vision device with the namespace `xmlns="http://www.hikvision.com/ver20/XMLSchema"`.
+
+**Input XML**
+```xml
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: Needs a blank line above code fence.

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 4]
```diff
@@ -1,3 +1,52 @@
...
+		<elevation>0</elevation>
+		<azimuth>450</azimuth>
+		<absoluteZoom>10</absoluteZoom>
+	</AbsoluteHigh>
+</PTZStatus>
+```
+A simple xpath query to fetch the Azimut value does not work as it does not adress the namespace.
+
+** Rules File**
+```php
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: Needs a blank line above code fence.

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 5]
```diff
@@ -1,3 +1,52 @@
...
+val azimuth = transform("XPATH", "/PTZStatus/AbsoluteHigh/azimuth/text()", testXml.toString)
+```
+
+To Address the namespace in the query there are two ways one full qualified path and a simple paht which may not work in complex XML.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: "To address ..."

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 6]
```diff
@@ -1,3 +1,52 @@
...
+
+**Rules File**
+```php
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: Need blank line above code fence.

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 7]
```diff
@@ -1,3 +1,52 @@
...
+// Full qualified
+val mytest = transform("XPATH", "/*[local-name()='PTZStatus'    and namespace-uri()='http://www.hikvision.com/ver20/XMLSchema']
+                                 /*[local-name()='AbsoluteHigh' and namespace-uri()='http://www.hikvision.com/ver20/XMLSchema']
+                                 /*[local-name()='azimuth'      and namespace-uri()='http://www.hikvision.com/ver20/XMLSchema']
+                                 /text()[1]", testXml.toString)
+logInfo('debug','mytest : ' + mytest )
+```
+
+**Value in mytest**
+```
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: Need blank line above code fence

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 8]
```diff
@@ -1,3 +1,52 @@
...
+450
+```
+
+## Further Reading
+For a [introduction](https://www.w3schools.com/xml/xpath_intro.asp) of XPath have a look at W3School.
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: Probably better to shorten to "An introduction to XPath".

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 9]
```diff
@@ -1,3 +1,52 @@
...
+A informative explanation of [common mistakes](https://qxf2.com/blog/common-xpath-mistakes/).
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: "An informative"

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 10]
```diff
@@ -16,16 +18,23 @@ Given a retrived XML e.g. from an HIK Vision device with the namespace `xmlns="h
...
+A simple xpath query to fetch the Azimut value does not work as it does not address the namespace.
 
 ** Rules File**
+
 ```php
 val azimuth = transform("XPATH", "/PTZStatus/AbsoluteHigh/azimuth/text()", testXml.toString)
 ```
 
-To Address the namespace in the query there are two ways one full qualified path and a simple paht which may not work in complex XML.
+To Address the namespace in the query there are two ways  and .
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: This line is no longer needed.

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 11]
```diff
@@ -1,3 +1,105 @@
...
+									 Temperature_xml.state.toString)
+									 
+    // post the new value to the Number Item
+    Temperature.postUpdate( newValue )
+ end
+```
+
+Now the resulting Number can also be used in the label to [change the color](https://docs.openhab.org/configuration/sitemaps.html#label-and-value-colors) or in a rule as value to compare.
+
+## Further Reading
```
Comment (9037568) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: The three statements in this section should be turned into 3 bulleted items.

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 12]
```diff
@@ -1,3 +1,105 @@
...
+		<elevation>0</elevation>
+		<azimuth>450</azimuth>
+		<absoluteZoom>10</absoluteZoom>
+	</AbsoluteHigh>
+</PTZStatus>
+```
+
+A simple xpath query to fetch the Azimut value does not work as it does not address the namespace.
+
+To Address the namespace in the query there are two ways  and .
```
Comment (triller-telekom) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: I think this sentence is a leftover and should be removed, right?

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 13]
```diff
@@ -1,3 +1,105 @@
...
+
+returns 
+
+```
+<azimuth>450</azimuth>
+```
+
+### In Setup
+
+**Item**
```
Comment (triller-telekom) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: maybe change it to ".items file"

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 14]
```diff
@@ -1,3 +1,105 @@
...
+
+```csv
+String  Temperature_xml "Temperature [JSONPATH([name()='PTZStatus']/*[name()='AbsoluteHigh']/*[name()='azimuth']/):%s °C]" {...}
+Number  Temperature "Temperature [%.1f °C]"
+```
+
+**Rule**
```
Comment (triller-telekom) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: maybe change it to ".rules file"

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 15]
```diff
@@ -1,3 +1,105 @@
...
+rule "Convert XML to Item Type Number"
+  when
+    Item Temperature_xml changed
+ then
+    // use the transformation service to retrieve the value
+	// Simple
+	val mytest = transform("XPATH", "/*[name()='PTZStatus']
+									 /*[name()='AbsoluteHigh']
+									 /*[name()='azimuth']
+									 /text()", testXml.toString)
```
Comment (triller-telekom) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: Where does "testXml" come from here?

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 16]
```diff
@@ -1,3 +1,105 @@
...
+  when
+    Item Temperature_xml changed
+ then
+    // use the transformation service to retrieve the value
+	// Simple
+	val mytest = transform("XPATH", "/*[name()='PTZStatus']
+									 /*[name()='AbsoluteHigh']
+									 /*[name()='azimuth']
+									 /text()", testXml.toString)  
+	// Full qualified
```
Comment (triller-telekom) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: Full**y** qualified

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 17]
```diff
@@ -1,3 +1,105 @@
...
+									 /*[local-name()='azimuth'      and namespace-uri()='http://www.hikvision.com/ver20/XMLSchema']
+									 /text()",
+									 Temperature_xml.state.toString)
+									 
+    // post the new value to the Number Item
+    Temperature.postUpdate( newValue )
+ end
+```
+
+Now the resulting Number can also be used in the label to [change the color](https://docs.openhab.org/configuration/sitemaps.html#label-and-value-colors) or in a rule as value to compare.
```
Comment (triller-telekom) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: value for comparison.

File: extensions/transform/org.eclipse.smarthome.transform.xpath/README.md [hunk 18]
```diff
@@ -83,20 +81,21 @@ rule "Convert XML to Item Type Number"
...
-									 Temperature_xml.state.toString)
+									 Temperature_xml.state.toString )
 									 
     // post the new value to the Number Item
     Temperature.postUpdate( newValue )
  end
 ```
 
-Now the resulting Number can also be used in the label to [change the color](https://docs.openhab.org/configuration/sitemaps.html#label-and-value-colors) or in a rule as value to compare.
+Now the resulting Number can also be used in the label to [change the color](https://docs.openhab.org/configuration/sitemaps.html#label-and-value-colors) or in a rule as value for compare.
```
Comment (triller-telekom) [inline extensions/transform/org.eclipse.smarthome.transform.xpath/README.md]: for compar**ison**.

**Pull request merged**